<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(45, 46, 48);</string>
  </property>
  <widget class="QWidget" name="gridLayoutWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>501</width>
     <height>501</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="container">
    <property name="leftMargin">
     <number>10</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <property name="bottomMargin">
     <number>10</number>
    </property>
    <property name="spacing">
     <number>9</number>
    </property>
    <item row="1" column="0">
     <widget class="QLabel" name="box_5">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>5</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="box_2">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>2</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLabel" name="box_3">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>3</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="3">
     <widget class="QLabel" name="box_4">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>4</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="box_9">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>10</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="box_13">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>9</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="3" column="3">
     <widget class="QLabel" name="box_16">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>15</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="3" column="2">
     <widget class="QLabel" name="box_15">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>14</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLabel" name="box_6">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QLabel" name="box_8">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>7</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLabel" name="box_10">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>11</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QLabel" name="box_11">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>12</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="box_1">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>1</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QLabel" name="box_7">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>6</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QLabel" name="box_14">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>13</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="2" column="3">
     <widget class="QLabel" name="box_12">
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>36</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(251, 251, 251);
border-radius: 15px;
color: rgb(45, 46, 48);</string>
      </property>
      <property name="text">
       <string>8</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="start_button">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>270</y>
     <width>90</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Yu Gothic</family>
     <pointsize>12</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(68, 173, 83);
color: rgb(255, 255, 255);</string>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
  </widget>
  <widget class="QPushButton" name="reset_button">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>270</y>
     <width>90</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Yu Gothic</family>
     <pointsize>12</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(68, 173, 83);
color: rgb(255, 255, 255);</string>
   </property>
   <property name="text">
    <string>Reset</string>
   </property>
  </widget>
  <widget class="QLabel" name="info">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>10</y>
     <width>280</width>
     <height>50</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>14</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>INPUT</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="set_button">
   <property name="geometry">
    <rect>
     <x>715</x>
     <y>125</y>
     <width>75</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Yu Gothic</family>
     <pointsize>12</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: #D3D3D3;
color: rgb(0,0,0);</string>
   </property>
   <property name="text">
    <string>Set</string>
   </property>
  </widget>
  <widget class="QLabel" name="info_1">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>170</y>
     <width>260</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>•  Input berisi 16 bilangan bulat</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="contoh">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>130</y>
     <width>200</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>8</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>ex: 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="info_2">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>200</y>
     <width>260</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>•  Input harus berada dalam range 0—15</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="info_3">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>230</y>
     <width>260</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>•  Input tidak boleh duplikat</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="input_box">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>70</y>
     <width>280</width>
     <height>50</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: #fbfbfb;
color: rgb(0,0,0);</string>
   </property>
  </widget>
  <widget class="QLabel" name="info_4">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>310</y>
     <width>260</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>15</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string> Is solvable ?</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="solvable">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>350</y>
     <width>130</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>•  Solvable</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="not_solvable">
   <property name="geometry">
    <rect>
     <x>640</x>
     <y>350</y>
     <width>130</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>•  Not solvable</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="n_steps">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>430</y>
     <width>40</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>15</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">color:rgb(45, 46, 48);
background-color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="info_6">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>430</y>
     <width>80</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>18</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>Steps</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="info_7">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>380</y>
     <width>170</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>12</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>Jumlah expanded node</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="n_expanded_nodes">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>380</y>
     <width>50</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>15</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">color:rgb(45, 46, 48);
background-color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="info_8">
   <property name="geometry">
    <rect>
     <x>680</x>
     <y>380</y>
     <width>40</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>12</pointsize>
     <underline>false</underline>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(45, 46, 48);
color:  rgb(251, 251, 251);</string>
   </property>
   <property name="text">
    <string>=</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="prev_button">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>435</y>
     <width>71</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Yu Gothic</family>
     <pointsize>12</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(68, 173, 83);
color: rgb(255, 255, 255);</string>
   </property>
   <property name="text">
    <string>Prev</string>
   </property>
  </widget>
  <widget class="QPushButton" name="next_button">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>435</y>
     <width>71</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Yu Gothic</family>
     <pointsize>12</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(68, 173, 83);
color: rgb(255, 255, 255);</string>
   </property>
   <property name="text">
    <string>Next</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
